### PR TITLE
[2.8] Add upgrade step to reset rel limit in existing charm metadata

### PR DESCRIPTION
**Pull request by achilleasa**
## Description of change

Before juju 2.8, the parser for charm metadata would impose a relation limit of 1 for `requirer` and `peer` relations when the charm metadata did not define any limit. This has been changed in 2.8 (see https://github.com/juju/charm/pull/309) so the limit could be correctly enforced in https://github.com/juju/juju/pull/11446.

However, the charm metadata stored by older controllers would still have a forced limit of 1 for requirer/peer relations. Thus, after an upgrade to 2.8 where relation limits are properly enforced, folks started getting limit exceeded errors when trying to add new relations.

To fix this, this PR introduces an upgrade step to reset the limits for the aforementioned endpoints to **0**.

## QA steps

```console
# Bootstrap a 2.7 controller
$ /snap/bin/juju version
2.7.8-focal-amd64

$ /snap/bin/juju bootstrap lxd-with-cache charm-upgrade-test --no-gui

# Deploy charms
$ /snap/bin/juju deploy telegraf
$ /snap/bin/juju deploy memcached
$ /snap/bin/juju deploy memcached another-memcached

# Check mongo: db.charms.find({}, {"meta":1}).pretty() 
# Observe the limit of 1 for requires/peer relations

# Upgrade to 2.8.1 (use snap here)
$ /snap/bin/juju upgrade-controller --agent-version 2.8.1
$ /snap/bin/juju upgrade-model

# You should hit the error now on the second command
$ /snap/bin/juju relate telegraf:memcached memcached
$ /snap/bin/juju relate telegraf:memcached another-memcached
ERROR cannot add relation "telegraf:memcached another-memcached:cache": establishing a new relation for telegraf:memcached would exceed its maximum relation limit of 1 (quota limit exceeded)

# Upgrade to 2.8.2 (use juju agent from PR)
$ juju upgrade-controller --build-agent
$ juju upgrade-model

# This should work now as expected
$  /snap/bin/juju relate telegraf:memcached another-memcached

$ juju status --relations
juju status --relations
...
Relation provider          Requirer                   Interface              Type         Message
another-memcached:cache    telegraf:memcached         memcache               subordinate  joining <----
another-memcached:cluster  another-memcached:cluster  memcached-replication  peer         joining
memcached:cache            telegraf:memcached         memcache               subordinate  joining <----
memcached:cluster          memcached:cluster          memcached-replication  peer         joining
```

## Bug reference

https://bugs.launchpad.net/juju/+bug/1887095

### CODE REVIEW

File: upgrades/steps_282.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package upgrades
+
+// stateStepsFor282 returns upgrade steps for Juju 2.8.2.
+func stateStepsFor282() []Step {
```
Comment (hmlanigan) [inline upgrades/steps_282.go:7]: Where is steps_282_test.go?  Looks like we stopped writing these tests for 2.8.x?

Comment (achilleasa) [reply upgrades/steps_282.go:7]: Yeah, the 2.8.1 ones where missing as well so I skipped them for 2.8.2. Not sure about the benefits of having these particular test suites... however, I did update the test in this package which verifies that the migration steps for 2.8.2 are registered for execution.

Comment (achilleasa) [reply upgrades/steps_282.go:7]: @hmlanigan Ignore my previous comment. I found the tests for 2.8.1 (the test file was misspelled :D). I pushed a commit with the missing tests + correct spelling for 2.8.1 test suite.

**Comment by achilleasa** [comment]
$$merge$$

**Pull request merged**